### PR TITLE
bump libocr; add context

### PR DESCRIPTION
**Pull request by jmank88**
https://smartcontract-it.atlassian.net/browse/BCF-2887

Supports:
- https://github.com/smartcontractkit/chainlink-automation/pull/304
- https://github.com/smartcontractkit/chainlink-data-streams/pull/52
- https://github.com/smartcontractkit/chainlink-feeds/pull/65
- https://github.com/smartcontractkit/chainlink-vrf/pull/47
- https://github.com/smartcontractkit/chainlink-cosmos/pull/401
- https://github.com/smartcontractkit/chainlink-solana/pull/567
- https://github.com/smartcontractkit/chainlink-starknet/pull/345
- https://github.com/smartcontractkit/chainlink/pull/14011

### CODE REVIEW

File: pkg/capabilities/consensus/ocr3/reporting_plugin_test.go [hunk 1]
```diff
@@ -401,22 +401,25 @@ func TestReportingPlugin_Reports_ShouldReportFalse(t *testing.T) {
...
 
-	ib := gotRep.Info
+	ib := gotRep.ReportWithInfo.Info
 	info, err := extractReportInfo(ib)
 	require.NoError(t, err)
 
 	assert.EqualExportedValues(t, info.Id, id)
 	assert.False(t, info.ShouldReport)
+
+	//TODO compare gotRep.TransmissionScheduleOverride
```
Comment (jmank88) [inline pkg/capabilities/consensus/ocr3/reporting_plugin_test.go:418]: - [ ] todo

File: pkg/loop/internal/relayer/relayer.go [hunk 2]
```diff
@@ -275,7 +275,9 @@ func WrapProviderClientConnection(providerType string, cc grpc.ClientConnInterfa
 	// for interoperability with legacy code.
 	switch providerType {
 	case string(types.Median):
-		return median.NewProviderClient(broker, cc), nil
+		pc := median.NewProviderClient(broker, cc)
+		pc.RmUnimplemented(context.Background())
```
Comment (jmank88) [inline pkg/loop/internal/relayer/relayer.go:279]: This is awkward. There is probably room for improvement here

File: pkg/capabilities/consensus/ocr3/reporting_plugin.go [hunk 3]
```diff
@@ -438,8 +439,12 @@ func (r *reportingPlugin) Reports(seqNr uint64, outcome ocr3types.Outcome) ([]oc
 				continue
 			}
 
-			rawReport, err = encoder.Encode(context.Background(), *mv)
+			rawReport, err = encoder.Encode(ctx, *mv)
 			if err != nil {
+				if cerr := ctx.Err(); cerr != nil {
```
Comment (ilija42) [inline pkg/capabilities/consensus/ocr3/reporting_plugin.go:444]: why this extra handling?

Comment (jmank88) [reply pkg/capabilities/consensus/ocr3/reporting_plugin.go:444]: To short circuit the loop

**Review by krehermann — APPROVED** [review]

**Pull request merged**